packages/legacy-credentials/src/Credential.ts

```diff
@@ -298,8 +298,14 @@ export function verifyWellFormed(
 ): void {
   verifyDataStructure(credential)
   verifyDataIntegrity(credential)
-
+  const credentialCTypeId = CType.hashToId(credential.claim.cTypeHash)
   if (ctype) {
+    if (credentialCTypeId !== ctype.$id) {
+      throw new SDKErrors.CTypeIdMismatchError(
+        ctype.$id,
+        credential.claim.cTypeHash
```

Comment:
and maybe that can be the `credentialCTypeId` as well then?

Reply:
https://github.com/KILTprotocol/sdk-js/pull/867/commits/56bc559147a83ccb16418dd20bb143af32726500